src/test/util/setup_common.cpp

```diff
@@ -159,7 +204,13 @@ BasicTestingSetup::~BasicTestingSetup()
     m_node.kernel.reset();
     SetMockTime(0s); // Reset mocktime for following tests
     LogInstance().DisconnectTestLogger();
-    fs::remove_all(m_path_root);
+    if (m_has_custom_datadir) {
+        // Only remove the lock file, preserve the data directory.
+        UnlockDirectory(m_path_lock, ".lock");
+        fs::remove_all(m_path_lock / ".lock");
```

Comment:
Should probably be `fs::remove` instead of `fs::remove_all`?

Reply:
Good catch, force-pushed d27e2d87b95b7982c05b4c88e463cc9626ab9f0a to fix.